crates/motionfile/src/motion_interpolator.rs

```diff
@@ -224,6 +238,22 @@ impl<T: Debug + Interpolate<f32>> MotionInterpolator<T> {
             keyframe.spline.set_initial_positions(position);
         }
     }
+
+    pub fn remaining_estimated_duration(&self) -> Duration {
+        match (
+            self.current_state.current_frame_index(),
+            self.current_state.current_frame_time_since_start(),
+        ) {
+            (Some(index), Some(time)) => Duration::saturating_sub(
+                self.frames[index..]
+                    .iter()
+                    .map(|frame| frame.spline.total_duration())
+                    .sum::<Duration>(),
+                time,
+            ),
+            _ => Duration::ZERO,
```

Comment:
use `None, None => Duration::ZERO` and let the other patterns panic with adequate error message since they cannot happen

Reply:
No longer relevant